packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1257,8 +1319,25 @@ export class TransactionController extends BaseController<
         delete txParams.gasPrice;
       }
 
+      if (!this.beforePublish(transactionMeta)) {
+        transactionMeta.status = TransactionStatus.approved;
+        this.updateTransaction(
+          transactionMeta,
+          'TransactionController#approveTransaction - Transaction approved',
+        );
```

Comment:
Do we not need to `return` here and ensure the signing logic still occurs?

It's more awkward here as the signing and publishing logic are all in one method so could be easier if we create a private `signTransaction` method?